features/release.feature

```diff
@@ -294,3 +294,42 @@ MARKDOWN
       """
       Attaching release asset `hello-1.2.0.tar.gz'...\n
       """
+
+    Scenario: Download a release asset.
+      Given the GitHub API server:
+        """
+        get('/repos/mislav/will_paginate/releases') {
+          json [
+            { url: 'https://api.github.com/repos/mislav/will_paginate/releases/123',
+              upload_url: 'https://api.github.com/uploads/assets{?name,label}',
+              tag_name: 'v1.2.0',
+              name: 'will_paginate 1.2.0',
+              draft: true,
+              prerelease: false,
+              assets: [
+                { url: 'https://api.github.com/repos/mislav/will_paginate/assets/9876',
+                  name: 'hello-1.2.0.tar.gz',
+                },
+              ],
+            },
+          ]
+        }
+        get('/repos/mislav/will_paginate/assets/9876') {
+          status 302
+          headers['Location'] = 'https://github-cloud.s3.amazonaws.com/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz'
+          ""
+        }
+        get('/releases/12204602/22ea221a-cf2f-11e2-222a-b3a3c3b3aa3a.gz') {
```

Comment:
@mislav I tried, but it doesn't seem to get called. afb17d8

Interestingly, this worked with real GH releases for a week or so after I opened this PR. Today, it no longer works - probably because of the host header, though I'm not totally sure. I verified cURL has a correct host header while following the redirects.